test/gui/tst_sharing/test.feature

```diff
@@ -98,44 +143,112 @@ Feature: Sharing
+        And user "Alice" has shared folder "Parent" on the server with user "Brian" with "read" permissions
+        And user "Alice" has shared file "textfile.txt" on the server with user "Brian" with "read" permissions
+        And user "Brian" has set up a client with default settings
+        When the user tries to overwrite the file "Parent/textfile.txt" with content "overwrite file inside a folder"
+        And the user tries to overwrite the file "textfile.txt" with content "overwrite file in the root"
+        Then as "Brian" the file "Parent/textfile.txt" on the server should have the content "file inside a folder"
+        And as "Brian" the file "textfile.txt" on the server should have the content "file in the root"
+        And as "Alice" the file "Parent/textfile.txt" on the server should have the content "file inside a folder"
+        And as "Alice" the file "textfile.txt" on the server should have the content "file in the root"
+
+
+    Scenario: sharee edits shared files and again try to edit after write permission is revoked
+        Given user "Alice" has uploaded file with content "ownCloud test text file" to "textfile.txt" on the server
+        And user "Alice" has created folder "FOLDER" on the server
+        And user "Alice" has uploaded file with content "some content" to "FOLDER/simple.txt" on the server
+        And user "Brian" has been created on the server with default attributes and without skeleton files
+        And user "Alice" has shared file "textfile.txt" on the server with user "Brian" with "all" permissions
+        And user "Alice" has shared folder "FOLDER" on the server with user "Brian" with "all" permissions
+        And user "Brian" has set up a client with default settings
+        When the user overwrites the file "textfile.txt" with content "overwrite ownCloud test text file"
+        And the user overwrites the file "FOLDER/simple.txt" with content "overwrite some content"
+        Then as "Brian" the file "textfile.txt" on the server should have the content "overwrite ownCloud test text file"
+        And as "Alice" the file "textfile.txt" on the server should have the content "overwrite ownCloud test text file"
+        And as "Brian" the file "FOLDER/simple.txt" on the server should have the content "overwrite some content"
+        And as "Alice" the file "FOLDER/simple.txt" on the server should have the content "overwrite some content"
+        When the user adds another account with
+            | server   | %local_server% |
+            | user     | Alice          |
+            | password | 1234           |
+        And the user removes permissions "edit" for user "Brian Murphy" of resource "textfile.txt" using the client-UI
+        And the user closes the sharing dialog
+        And the user removes permissions "edit" for user "Brian Murphy" of resource "FOLDER" using the client-UI
```

Comment:
is this test about changing the sharing from the UI or about the overwriting?
Maybe we could have those steps done by the API?

Reply:
The scenario is to do `overwrite` -> `change permissions` -> `try to overwrite` using the client UI.
Any suggestions?